### DISCUSSION TITLE
Host custom static files in meilisearch

### DISCUSSION THREAD

**Discussion by joepio** [meilisearch/product#610] [Feedback & Feature Proposal]
Most of the time somebody uses Meilisearch, they'll also want to host some static files with it. Now of course they can then spin up an nginx server or whatever... But that's one more application you need to run and maintain.

Meilisearch already hosts some static files. It already uses actix. What if users could add their own files to a `user-static` and let Meilisearch host these? It's a bit of scope creep, and probably a bad idea, so let's explore some considerations before doing anything stupid:

- Users probably want to have a custom `index.html`, so we should be able to set things on root. This will break the dashboard. I don't think that's a horrible problem, or trade-off. 
- You could move the `dashboard` handler to `/dashboard`, and only redirect if no `index.html` is found in the `user-static` folder.
- How about supporting subdirectories? If you support these, from root, then you could have collisions with meilisearch paths. Some users could have folders that will no longer be available after some update. 

EDIT: [PR here](https://github.com/meilisearch/meilisearch/pull/3478)

**Comment by macraig** [comment]
Hi @joepio, thank you for taking the time and leaving your feedback! It feels like the main pain point here is around making HTTPS easier to set up. I see you've already proposed a solution for that in https://github.com/meilisearch/product/discussions/609, so I'll close this one for now.

**Reply by joepio** [reply]
I don't really see how these issues are the same! They vaguely relate, as both file management and HTTPS are supported by Nginx, but I think they are separate in principle.

**Reply by macraig** [reply]
My bad, I was reading the linked PR that was referencing the other discussion and thought they might be solving the same pain point. I now see that the PR relates to both of them but they are in fact separate concerns. Even though this doesn't fit into our current priorities, I'll reopen the discussion for future consideration. Once again thank you for your feedback!
